perun/view_diff/sankey_incremental/run.py

```diff
@@ -0,0 +1,674 @@
+"""Sankey difference of the profiles
+
+The difference is in form of:
+
+cmd       | cmd
+workload  | workload
+collector | kernel
+
+| ---|          |======|
+     |-----|====
+|---|          |------|
+
+"""
+
+from __future__ import annotations
+
+# Standard Imports
+from operator import itemgetter
+from typing import Any, Literal, Type, Callable
+from collections import defaultdict
+from dataclasses import dataclass
+import sys
+
+# Third-Party Imports
+import array
+import click
+import jinja2
+import progressbar
+
+# Perun Imports
+from perun.profile import convert
+from perun.profile.factory import Profile
+from perun.templates import filters
+from perun.utils import log, mapping
+from perun.utils.common import diff_kit, common_kit
+from perun.view_diff.flamegraph import run as flamegraph_run
+
+
+def singleton_class(cls: Type[Any]) -> Callable[[], Config]:
+    """Helper class for creating singleton objects"""
+    instances = {}
+
+    def getinstance() -> Config:
+        """Singleton instance"""
+        if cls not in instances:
+            instances[cls] = cls()
+        return instances[cls]
+
+    return getinstance
+
+
+@singleton_class
+class Config:
+    """Singleton config for generation of sankey graphs
+
+    :ivar trace_is_inclusive: whether then the amounts are distributed among the whole traces
+    """
+
+    def __init__(self) -> None:
+        """Initializes the config
+
+        By default we consider, that the traces are not inclusive
+        """
+        self.trace_is_inclusive: bool = False
+        self.top_n_traces: int = 5
+
+
+class ColorPalette:
+    """Colour palette is a static object for grouping colours used in visualizations"""
+
+    Baseline: str = "rgba(49, 48, 77, 0.4)"
+    Target: str = "rgba(255, 201, 74, 0.4)"
+    Increase: str = "rgba(255, 0, 0, 0.7)"
+    Decrease: str = "rgba(0, 255, 0, 0.7)"
+    Equal: str = "rgba(0, 0, 255, 0.7)"
+    DarkTarget: str = "rgba(255, 201, 74, 1)"
+    DarkBaseline: str = "rgba(49, 48, 77, 1)"
+    DarkIncrease: str = "#ea5545"
+    DarkDecrease: str = "#87bc45"
+    DarkEqual: str = "#27aeef"
+    Highlight: str = "rgba(0, 0, 0, 0.7)"
+    NoHighlight: str = "rgba(0, 0, 0, 0.2)"
```

Comment:
Doesn't the `sankey/run.py` define a ColorPalette class as well? It seems at least some colors are the same? Maybe consider merging the classes.

Reply:
Good point, I actually want to extract all colours to isolate package, but for now, I extracted this to WebColorPalette class in perun/utils/structs.py